autopilot/accounts.go

```diff
@@ -168,6 +168,17 @@ func (a *accounts) refillWorkerAccounts(w Worker) {
 				return fmt.Errorf("drift on account is too large - not funding")
 			}
 
+			// Check if a resync is needed.
+			if account.RequiresSync {
```

Comment:
You have to either return the balance after syncing or update the `account` here before checking the balance for refill?